### DISCUSSION TITLE
Whether or not to include docstrings in testing functions

### DISCUSSION THREAD

**Discussion by jacobcook1995** [ImperialCollegeLondon/virtual_rainforest#61] [General]
At the moment, flake8 raises if docstrings are not included in the testing. This covers both the testing functions themselves (rule D103) and docstrings at the top of each script (rule D100). We probably should decide whether we want doc strings for either of these cases and if not disable them for testing scripts using
```cfg
per-file-ignores =
    test/test_*.py: D100, D103
```
in the `setup.cfg` file.

**Comment by davidorme** [comment]
I think we've kind of decided on this, in favour of including docstrings? Basically, those docstrings are a logical place to put a summary on how a test works or what it is doing, and those are sometimes going to be really valuable. That means we have to put trivial docstrings on most functions, but that isn't a huge deal?

We definitely don't want to *have* to document arguments in tests. That would be a pain but at the moment, it looks like argument documentation checks only kick in once you have an `Args:` section.
